src/rpm_ostree/mod.rs

```diff
@@ -20,10 +21,18 @@ pub struct Release {
 
 impl Release {
     /// Builds a `Release` object from a Cincinnati node.
-    pub fn from_cincinnati(node: Node) -> Self {
-        Self {
+    pub fn from_cincinnati(node: Node) -> Fallible<Self> {
+        let payload_is_checksum = node
+            .metadata
+            .get(SCHEME_KEY)
+            .map(|v| v == CHECKSUM_SCHEME)
+            .unwrap_or(false);
+        ensure!(payload_is_checksum, "unexpected non-checksum payload");
```

Comment:
Should we include the actual scheme we got instead here?

Reply:
We can. There is the further case where there is no scheme at all, though.

Reply:
Consider it optional. Not blocking on this.